plugins/package-managers/bazel/src/main/kotlin/BazelModel.kt

```diff
@@ -31,8 +31,18 @@ private val json = Json { ignoreUnknownKeys = true }
  */
 @Serializable
 internal data class Lockfile(
-    val flags: Flags? = null
+    // The flags containing the registry URLs (Bazel < 7.2.0).
+    val flags: Flags? = null,
+
+    // The registry URLs of the project's dependencies packages  (Bazel >= 7.2.0).
+    val registryFileHashes: Map<String, String>? = null
 ) {
+    init {
+        require((flags != null && registryFileHashes == null) || (flags == null && registryFileHashes != null)) {
+            "Exactly one of 'flags' and 'registryFileHashes' must be set."
+        }
+    }
```

Comment:
Previously, we've been tolerating absent `flags`. Should we continue to have the same tolerance and simply drop the check?

Reply:
I don't think so: yes, we were tolerating i., but it was a nasty silent error: you could find yourself without registry which mean the package manager wouldn't be able to resolve any source code.

Reply:
I just digged a bit deeper and found `BazelTest` which copies a definition file
to a temp dir before analyzing it, see [1].

If it would not do the copying, then the test would now fail, because the lockfile contains both, `flags` and `registryFileHashes`. So, it indeed seems to be possible that both are defined.

Can you shed a light on
1. How that lockfile has been created
2. Why this temp dir / copy approach has been choosen in that test?

[1] https://github.com/oss-review-toolkit/ort/blob/39a638e55139598757c6f60cc8c830b49e0e2b1e/plugins/package-managers/bazel/src/test/kotlin/BazelTest.kt#L36.

Reply:
1. Manually, when I was not fully understanding what Bazel was doing with the lock files :smile: 

> flags and registryFileHashes. So, it indeed seems to be possible that both are defined.

Wrong. When you run `bazel mod graph` in the test project directory, the lock file is automatically refreshed with the `bazel` version you have installed (since there is no `bazelversion` file in this project).
So if I have `7.2.0` installed, the lock file is regenerated without any `flags`.

2. Because it was making sense at the time :shrug:  Feel free to replace it with a functional test.

Reply:
Could we fix-up this test in a PR before this PR, or alternatively in a preparing commit in this PR, so this confusion is removed? (e.g. the mistake in the lockfile needs to be fixed) 

If it makes sense to keep the test, it would probably be converted to a functional test.
If not, the assets could just be deleted along with the test.